collections_hierarchy/mixins.py

```diff
@@ -1,54 +1,81 @@
+from copy import deepcopy
+
 class ComparableMixin(object):
+    #Implements comparison functionality. Common for List and Dict
     def __eq__(self, other):
-        pass
+        data1 = getattr(self, self.DATA_ATTR_NAME) #Get the value of the property that stores the data
+        data2 = getattr(other, other.DATA_ATTR_NAME)
+        
+        if isinstance(data1, dict):
+            return data1 == data2
+        else:
+            return sorted(data1) == sorted(data2)
+        
     def __ne__(self, other):
-        # Relies in __eq__
-        pass
-
+        return not self.__eq__(other)
```

Comment:
I think 'self == other' is a bit more readable. That's why you define this magic method.